merk/src/merk/restore.rs

```diff
@@ -121,14 +122,14 @@ impl<'db, S: StorageContext<'db>> Restorer<S> {
     /// Verifies the trunk then writes its data to the RocksDB.
     fn process_trunk(&mut self, ops: impl IntoIterator<Item = Op>) -> Result<usize, Error> {
         let (trunk, height) = verify_trunk(ops.into_iter().map(Ok)).unwrap()?;
-        if trunk.key() != self.expected_root_key {
```

Comment:
why remove this?

Reply:
It was originally trunk.hash().unwrap() != self.expected_root_hash, but this was comparing combined hash with underlying merk hash.
I am now checking if it's a non-root tree and generating the combined hash before comparing.